documentation/src/docs/streamcuts.md

```diff
@@ -0,0 +1,92 @@
+<!--
+Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+-->
+# Working with Pravega: StreamCuts
+
+This section describes about `StreamCut`s and how they can be used with streaming clients and batch clients.
+Pre-requisites: You should be familiar with [Pravega Concepts](http://pravega.io/docs/latest/pravega-concepts/).
+
+## Definition
+
+A Pravega stream is formed by one or multiple parallel segments for storing/reading events. A Pravega stream
+is elastic, which means that the number of parallel segments may change along time to accommodate
+fluctuating workloads. That said, a `StreamCut` represents a consistent position in the stream. It contains
+a set of segment and offset pairs for a single stream which represents the complete keyspace at a given
+point in time. The offset always points to the event boundary and hence there will be no offset pointing to
+an incomplete event.
+
+The `StreamCut` representing the tail of the stream (with the newest event) is an ever changing one since
+events can be continuously added to the stream and the `StreamCut` pointing to the tail of the stream with
+newer events would have a different value. Similarly the `StreamCut` representing the head of the
+stream (with the oldest event) is an ever changing one as the stream retention policy could truncate the stream
+and the `StreamCut` pointing to the head of the stream post truncation would have a different value.
+```StreamCut.UNBOUNDED``` is used to represent such a position in the stream and the user can use it to
+specify this ever changing stream position (both head and tail of the stream).
+
+It should be noted that `StreamCut`s obtained using the streaming client and batch client can be used
+interchangeably.
+
+## StreamCut with Streaming clients
```

Comment:
You may want to refer to reads explicitly in this title as the discussion is about readers. I'm saying this because writers are also clients, and this is not about writers.

Reply:
yes, it is about Readers, updated the title.